src/sqlfluff/core/linter/linted_file.py

```diff
@@ -315,88 +302,66 @@ def fix_string(self) -> Tuple[Any, bool]:
             elif (
-                enriched_patch.source_slice.start == enriched_patch.source_slice.stop
-                and enriched_patch.source_slice.start == local_raw_slices[0].source_idx
+                patch.source_slice.start == patch.source_slice.stop
+                and patch.source_slice.start == local_raw_slices[0].source_idx
             ):
                 linter_logger.info(
                     "      * Keeping insertion patch on slice boundary: %s",
-                    enriched_patch,
-                )
-                filtered_source_patches.append(enriched_patch)
-                dedupe_buffer.append(enriched_patch.dedupe_tuple())
-            # If it's ONLY templated then we should skip it.
-            elif "literal" not in local_type_list:  # pragma: no cover
-                linter_logger.info(
-                    "      - Skipping patch over templated section: %s", enriched_patch
-                )
-            # If it's an insertion (i.e. the string in the pre-fix template is '') then
-            # we won't be able to place it, so skip.
-            elif not enriched_patch.templated_str:  # pragma: no cover TODO?
-                linter_logger.info(
-                    "      - Skipping insertion patch in templated section: %s",
-                    enriched_patch,
-                )
-            # If the string from the templated version isn't in the source, then we
-            # can't fix it.
-            elif (
-                enriched_patch.templated_str not in enriched_patch.source_str
-            ):  # pragma: no cover TODO?
-                linter_logger.info(
-                    "      - Skipping edit patch on templated content: %s",
-                    enriched_patch,
+                    patch,
                 )
-            else:  # pragma: no cover
+                filtered_source_patches.append(patch)
+                dedupe_buffer.append(patch.dedupe_tuple())
+            else:
+                # We've got a situation where the ends of our patch need to be
+                # more carefully mapped. Likely because we're greedily including
+                # a section of source templating with our fix and we need to work
+                # around it gracefully.
+
                 # Identify all the places the string appears in the source content.
-                positions = list(
-                    findall(enriched_patch.templated_str, enriched_patch.source_str)
-                )
+                positions = list(findall(patch.templated_str, patch.source_str))
                 if len(positions) != 1:
-                    linter_logger.debug(
+                    # NOTE: This section is not covered in tests. While we
+                    # don't have an example of it's use (we should), the
+                    # code after this relies on there being only one
+                    # instance found - so the safety check remains.
+                    linter_logger.debug(  # pragma: no cover
```

Comment:
Possible future work: Could we avoid the possibility of this occurring by allowing (requiring) rules to provide additional position info when making changes in templated areas?